transport/transporters/kafka/utils/kafka.go

```diff
@@ -0,0 +1,37 @@
+/*
+  Copyright 2023 Nextdoor.com, Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package utils
+
+type KafkaPartitionMethod int
+
+const (
+	KAFKA_PART_TXN        KafkaPartitionMethod = iota // Uses each msgs's txn to select kafka partition
+	KAFKA_PART_BATCH                                  // All messages in a batch go to the same random partition
+	KAFKA_PART_RANDOM                                 // Messages are randomly assigned a partition by Sarama
+	KAFKA_PART_TXN_CONST                              // Similar to KAFKA_PART_TXN but should only be used for testing
+	KAFKA_PART_TABLE_NAME                             // Partitions by message table name
```

Comment:
I'm curious, do you think it would make sense to enforce restrictions for combinations between kafka partitioning and [bifrost partitioning](https://github.com/Nextdoor/pg-bifrost/blob/main/partitioner/partitioner.go#L33-L36)? Say for example, using `KAFKA_PART_TABLE_NAME` with `PART_METHOD_TXN_BUCKET`

Reply:
That's not a bad idea but it'll require sitting down and enumerating the valid ways and their use cases. I struggled verbalizing certain configurations when I was writing this initially. FWIW I initially made this be 100% dependent on the bifrost partition method but that turned out to not be flexible enough and the overlap isn't 1:1 so it created some weird cases.